markdown/extensions/toc.py

```diff
@@ -142,22 +142,30 @@ def iterparent(self, root):
         for parent in root.iter():
             for child in parent:
                 yield parent, child
+                
+    def iter_allowed_parent(self, node):
```

Comment:
Why did you not just modify iterparent?  Now we have iterparent which is no longer and used, and then iter_allowed_parent.  This should be consolidated into one function, and if you have dead code, it should be removed.

Reply:
Because it's public, and I was not sure that it's not used anywhere. But if it's not the case, it's better to simply replace it.

Reply:
Fixed

Reply:
If it isn't overriding a function in the base class, it probably isn't used externally.